internal/transaction/manager.go

```diff
@@ -0,0 +1,28 @@
+// MIT License. Copyright (c) 2020 CQFN
+// https://github.com/cqfn/degitx/blob/master/LICENSE
+
+// Package transaction provides API interfaces and implementations
+// of atomic transaction for DeGitX. The workflow and research are
+// explained in the white-paper. The brief description of files
+// in this package:
+//  - hook.go - interfaces for API of git reference-transaction hook
+//    exposed for RM.
+//  - manager.go - transaction manager (TM) API for resource manager (RM).
+//  - meta.go - transaction metadata types, such as votes, scopes, identifiers.
+//  - paxos.go - Paxos-commit interfaces, such as acceptors and proposers.
+//  - rm.go - resource manager API for hook and TM.
+package transaction
+
+import (
+	"context"
+
+	"cqfn.org/degitx/locators"
+)
+
+// Manager of the transaction
+type Manager interface {
+	// Begin a transaction, update votes for RMs
```

Comment:
I don't get this interface. Who will begin a transaction? a RM? and who will Finish? and why we pass `Votes` in `Begin`?

Reply:
@stepanvalyavskiy I'll prepare a diagram for the white-paper soon (similar to paper diagram) to explain this. RM begins the transaction and sends all known votes to the TM. From the RM view, we can't be sure that the transaction was started or not, so we always have to begin the transaction assuming it was not started yet, and TM have to manage this case and update transaction votes if the transaction was started already.

Reply:
Now it's clear from comments